File: backend/src/main/java/com/pgvaale/backend/entity/Tiffin.java
package com.pgvaale.backend.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;

/** TIFFIN ENTITY */
@Entity
@Table(name = "tiffins")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Tiffin extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "tiffin_seq")
    @SequenceGenerator(name = "tiffin_seq", sequenceName = "tiffin_sequence", initialValue = 3000, allocationSize = 1)
    private Long id;

    @NotBlank
    @Size(min = 10, max = 10)
    private String phoneNumber;

    @NotBlank
    @Size(min = 12, max = 12)
    private String aadhaar;

    private double price;

    private String foodCategory; // "Veg" or "Non-Veg"

    private String region;

    private String maidAddress;
    
    @Builder.Default
    private boolean approved = false; // Admin approval status
} 

File: PGVaaleDotNetBackend/Entities/Tiffin.cs
using System.ComponentModel.DataAnnotations;

namespace PGVaaleDotNetBackend.Entities
{
    public class Tiffin : BaseEntity
    {
        [Required]
        [StringLength(10, MinimumLength = 10)]
        public string PhoneNumber { get; set; } = string.Empty;

        [Required]
        [StringLength(12, MinimumLength = 12)]
        public string Aadhaar { get; set; } = string.Empty;

        [Required]
        public double Price { get; set; }

        [Required]
        public string FoodCategory { get; set; } = string.Empty; // "Veg" or "Non-Veg"

        [Required]
        public string Region { get; set; } = string.Empty;

        [Required]
        public string MaidAddress { get; set; } = string.Empty;

        public bool Approved { get; set; } = false; // Admin approval status
    }
}